Tests/base/NSDictionary/sort.m

```diff
@@ -0,0 +1,43 @@
+#import "Testing.h"
+#import "ObjectTesting.h"
+#import <Foundation/NSAutoreleasePool.h>
+#import <Foundation/NSDictionary.h>
+#import <Foundation/NSValue.h>
+#import <Foundation/NSString.h>
+
+int main()
+{
+  NSAutoreleasePool   *arp = [NSAutoreleasePool new];
+
+  NSArray* values = [NSArray arrayWithObjects:
+        [NSNumber numberWithFloat:2.0],
+        [NSNumber numberWithFloat:1.0],
+        [NSNumber numberWithFloat:3.0],
+        [NSNumber numberWithFloat:4.0],
+        nil];
+
+  NSArray* keys = [NSArray arrayWithObjects:
+        @"shouldSortToSecond",
+        @"shouldSortToFirst",
+        @"shouldSortToThird",
+        @"shouldSortToFourth",
+        nil];
+
+  NSDictionary *d = [NSDictionary dictionaryWithObjects:values forKeys:keys];
+  NSArray* keysOrderedByKeyedValue = [d keysSortedByValueUsingComparator:
+                      ^NSComparisonResult(id obj1, id obj2) {
```

Comment:
You should protect this test with a HAS_BLOCK (or what ever this is called) guard, as gcc won't be able to compile this.